### PR TITLE
feat: Update dependencies in hale-platform to resolve security issues

### PR DESCRIPTION
**Pull request by emanuelaepure10**
update slf4jAndLogback from 1.0.13 to 1.2.12

ING-4067

### CODE REVIEW

**Comment by emanuelaepure10** [comment]
retest this please

File: build.gradle [hunk 1]
```diff
@@ -91,7 +91,7 @@ include {
 	}
 
 	from('modules/shared/logging.gradle') {
-		slf4jAndLogback '1.7.10', '1.0.13'
+		slf4jAndLogback '1.7.25', '1.0.13'
```
Comment (stempler) [inline build.gradle:94]: Did changing logback to `1.2.12` cause any problems?

File: build.gradle [hunk 2]
```diff
@@ -321,7 +321,7 @@ platform {
 //	bundle 'com.ning:async-http-client:1.8.15'
 
 	// quartz scheduler
-	bundle 'org.quartz-scheduler:quartz:1.7.3'
+	bundle 'org.quartz-scheduler:quartz:2.4.0-rc1'
```
Comment (stempler) [inline build.gradle:324]: Any specific reason for using a release candidate version?
Would be great to add that as comment.

File: build.gradle [hunk 3]
```diff
@@ -638,7 +638,7 @@ platform {
 	}
 	bnd group: 'ru.yandex.qatools.allure', name: 'allure-java-adaptor-api', {
 		// tika exported package version does not seem to resemble bundle version
-		instruction 'Import-Package', 'org.apache.tika.*;version="[1.0.0,2.0.0)",*'
+		instruction 'Import-Package', 'org.apache.tika.*;version="[1.28.4,2.9.1)",*'
```
Comment (stempler) [inline build.gradle:641]: This only impacts the `Import-Package` instructions, not the dependency version used.

File: build.gradle [hunk 4]
```diff
@@ -297,10 +297,10 @@ platform {
 	bundle 'asm:asm-analysis:3.3.1'
 
 	// ivy with at least 2.4 because of bug in IvySettings
-	bundle 'org.apache.ivy:ivy:2.4.0'
+	bundle 'org.apache.ivy:ivy:2.5.2'
 
 	// web stuff
-	bundle 'commons-fileupload:commons-fileupload:1.3.1'
+	bundle 'commons-fileupload:commons-fileupload:1.5'
```
Comment (stempler) [inline build.gradle:303]: Generally, even better than updating is probably removing stuff from the platform that is no longer needed. 
These dependencies listed here under "web stuff" are candidates for removal if they are in fact no longer used. There were some hale server components that were removed some time ago, that might have used these.
Quite certain that at least `openid4java`, `recaptcha4j` and anything related to Wicket can be removed.

**Review by stempler — COMMENTED** [review]
Did you perform any tests using an update site your changes for hale studio? What were the results?

Generally the main part of the process of updating/changing the dependencies in the platform is testing these changes in hale studio and applying the related changes/fixes the both the platform and hale studio.

Before merging changes here we need to make sure they have been validated with hale studio. Otherwise we end up with a state in hale-platform that someone else could not work with without resolving the related issues first.

File: build.gradle (hunk 2 above)
Comment (emanuelaepure10) [reply build.gradle:324]: Good point. Thank you
i was just blind not to see that the 2.3.2, which is just the last version of 2.3 is without vulnerabilities.

File: build.gradle (hunk 4 above)
Comment (emanuelaepure10) [reply build.gradle:303]: I'll remove them all 3:
```
bundle 'commons-fileupload:commons-fileupload:1.5'
bundle 'org.openid4java:openid4java:0.9.8'
bundle 'net.tanesha.recaptcha4j:recaptcha4j:0.0.8'
```

As for wicket I found that is pretty used, there is an entire shared module
```
from('modules/shared/wicket.gradle') {
	wicket(wicketVersion, [
		'wicket-core',
		'wicket-request',
		'wicket-util',
		'wicket-spring',
		'wicket-extensions'
	])
}
```
`bundle 'org.codehaus.groovy.modules.http-builder:http-builder:0.7.1'`

Can you confirm the removal please?

File: build.gradle (hunk 3 above)
Comment (emanuelaepure10) [reply build.gradle:641]: I see, but I cannot find any other place where tika has been used.

File: build.gradle (hunk 1 above)
Comment (emanuelaepure10) [reply build.gradle:94]: Why should I change it to a lower version?

**Comment by emanuelaepure10** [comment]
> Did you perform any tests using an update site your changes for hale studio? What were the results?

I haven't. As I couldn't conclude successfully the process of updating the resource bundles.
To have an update site changed with this changes I should have a commit in bnb platform. Can I just create a Pr for testing proposes?

File: build.gradle (hunk 1 above)
Comment (stempler) [reply build.gradle:94]: The first version is the SLF4J version, with you changes at `1.7.25`, the second version is the logback version, now at `1.0.13`.

File: build.gradle (hunk 4 above)
Comment (stempler) [reply build.gradle:303]: On `openid4java`, `recaptcha4j` and everything Wicket I'm sure.

I can't tell you for sure for the others, ultimately you need to test the changes with hale studio. As a first hint you could try to see if the bundles appear in a built version of hale studio or in the Tests or Infocenter products.

**Comment by stempler** [comment]
@emanuelaepure10 

> To have an update site changed with this changes I should have a commit in bnb platform. Can I just create a Pr for testing proposes?

Do you mean hale-build-support?
If you want to do tests with commits there then please create a fork of the repository in GitLab and push your changes to a branch there. That way we can avoid changes used for testing to make it into the main repository, which is already quite big.
A faster way is usually still to test directly in Eclipse. What I always do is the following:

1. Perform the hale-platform build
2. Rename the `updatesite` folder in `hale-platform/build` to something unique (e.g. append date and an increasing number each time), to avoid any issues with Eclipse caching for the location
3. Replace the target platform update site in the `.target` file with the local renamed updatesite folder (`file:/` url) and load the target platform from there
4. Validate `HALE.product` to look for problems and look into Target Platform state view for details

**Comment by emanuelaepure10** [comment]
1. build successful
2. rename folder to `updatesite20231107`
3. change `.target` to
```
<repository location="file:/C:/Users/EmanuelaEpure/Documents/GitHub/hale-platform-update/hale-platform/build/updatesite20231107"/>
<unit id="eu.esdihumboldt.hale.platform.feature.group" version="4.1.0.i20231107"/>
```
Not sure how to change the version...if I just change the date I get
image
@stempler also with a 5 i still see the problem
image
Any thoughts on what should I change?
Thank you

**Comment by stempler** [comment]
> Any thoughts on what should I change?

@emanuelaepure10 When you use the Eclipse editor for the target file (not editing the source XML) you should be able to edit the update site entry, and in the dialog that opens then select the item that is actually contained in the update site.

**Comment by emanuelaepure10** [comment]
@stempler Thank you.
I wrote the later update...that I have forgot to select the HALE target platform libraries.

File: build.gradle (hunk 4 above)
Comment (emanuelaepure10) [reply build.gradle:303]: I would need more details about what you mean with
"the bundles appear in a built version of hale studio or in the Tests or Infocenter products".

for example if I remove the [line related to bundle 'org.openid4java:openid4java:0.9.8'](https://github.com/halestudio/hale-platform/blob/bd60c74073c87613b5ff6b35296e808dd59d4c8e/build.gradle#L305)
I get an error for the [bundleFeatures](https://github.com/halestudio/hale-platform/blob/bd60c74073c87613b5ff6b35296e808dd59d4c8e/build.gradle#L939) and I see that the org.openid4java is used in [hale studio here ](https://github.com/halestudio/hale/blob/7e35a00782be84bd85ccc71d007061ee31a30764/build/bundleArtifactMap.json#L222)

**Comment by emanuelaepure10** [comment]
Open points where I would need help to solve them:

1. upgrading jetty from [jettyVersion = '9.2.1.v20140609'](https://github.com/halestudio/hale-platform/blob/bd60c74073c87613b5ff6b35296e808dd59d4c8e/modules/jetty-support/jetty-support.gradle#L3) to  ''9.4.53.v20231009'. Also [this in the bundleArtifactMap.json ](https://github.com/halestudio/hale/blob/7e35a00782be84bd85ccc71d007061ee31a30764/build/bundleArtifactMap.json#L229)should be changed?

2. remove `commons-fileupload`, `openid4java`, `recaptcha4j `and everything related to `Wicket`

3. Upgrade (or maybe follow up tickets)
```
modules/shared/orientdb.gradle
modules/shared/spring.gradle
modules/shared/postgis.gradle
```
which are giving problems.

4. How to solve the following:

> 4.1. // force package export versions for hsqldb
> 	bnd group: 'org.hsqldb', name: 'hsqldb', {
> 		instruction 'Export-Package', "*;version=$version"
> }
> 4.2. // force correct package export versions for sqlite-jdbc
> bnd group: 'org.xerial', name: 'sqlite-jdbc', {
> 		instruction 'Export-Package', "*;version=$version"
> }
>  4.3.// make OSGi JDBC service import optional
>     bnd group: 'com.h2database', name: 'h2', {
>       optionalImport 'org.osgi.service.jdbc'
>     }
> 4.4.// security
> 	bundle 'org.jasig.cas:cas-client:3.1.10'

Thank you

File: build.gradle (hunk 4 above)
Comment (stempler) [reply build.gradle:303]: > I would need more details about what you mean with
"the bundles appear in a built version of hale studio or in the Tests or Infocenter products".

The HALE product is feature based so it is not easy to determine directly from the product file, other options are:

- inspect the list of bundles shipped with a built version of hale studio (create or download build, check `plugins` folder)
- another way is possibly creating a run configuration from the product file and editing the run configuration to check for included bundles (that also allows for easy filtering)

The products Tests and Infocenter are plugin based, so they can be easily checked if a plugin is included or not:

- https://github.com/halestudio/hale/blob/master/common/plugins/eu.esdihumboldt.hale.common.test/Tests.product
- https://github.com/halestudio/hale/blob/master/doc/plugins/eu.esdihumboldt.hale.doc.application/Infocenter.product

> I see that the org.openid4java is used in hale studio here (https://github.com/halestudio/hale/blob/7e35a00782be84bd85ccc71d007061ee31a30764/build/bundleArtifactMap.json#L222)

This Json file supports in mapping from bundle names to Maven artifacts. Something being present there does not mean it is being used, only that it was used at some point.

File: build.gradle [hunk 5]
```diff
@@ -90,8 +90,8 @@ include {
 		]
 	}
 
-	from('modules/shared/logging.gradle') {
-		slf4jAndLogback '1.7.10', '1.0.13'
+	from('modules/logging.gradle') {
+		slf4jAndLogback '1.7.36', '1.0.13'
```
Comment (stempler) [inline build.gradle:94]: Can you please again include the update to logback from `1.0.13` to `1.2.12`? Thank you

**Review by stempler — APPROVED** [review]

**Pull request merged**